File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/WeatherLegCalculation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class WeatherLegCalculation : ISerialized
    {
        public WeatherLegCalculation(XmlNode xmlNode)
        {
            XmlNode settlementLevelNode = xmlNode.SelectSingleNode("settlementLevel");
            
            if (settlementLevelNode != null)
            {
                if (settlementLevelNode.Attributes["href"] != null || settlementLevelNode.Attributes["id"] != null) 
                {
                    if (settlementLevelNode.Attributes["id"] != null) 
                    {
                        settlementLevelIDRef_ = settlementLevelNode.Attributes["id"].Value;
                        WeatherSettlementLevelEnum ob = new WeatherSettlementLevelEnum(settlementLevelNode);
                        IDManager.SetID(settlementLevelIDRef_, ob);
                    }
                    else if (settlementLevelNode.Attributes["href"] != null)
                    {
                        settlementLevelIDRef_ = settlementLevelNode.Attributes["href"].Value;
                    }
                    else
                    {
                        settlementLevel_ = new WeatherSettlementLevelEnum(settlementLevelNode);
                    }
                }
                else
                {
                    settlementLevel_ = new WeatherSettlementLevelEnum(settlementLevelNode);
                }
            }
            
        
            XmlNode referenceLevelEqualsZeroNode = xmlNode.SelectSingleNode("referenceLevelEqualsZero");
            
            if (referenceLevelEqualsZeroNode != null)
            {
                if (referenceLevelEqualsZeroNode.Attributes["href"] != null || referenceLevelEqualsZeroNode.Attributes["id"] != null) 
                {
                    if (referenceLevelEqualsZeroNode.Attributes["id"] != null) 
                    {
                        referenceLevelEqualsZeroIDRef_ = referenceLevelEqualsZeroNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(referenceLevelEqualsZeroNode);
                        IDManager.SetID(referenceLevelEqualsZeroIDRef_, ob);
                    }
                    else if (referenceLevelEqualsZeroNode.Attributes["href"] != null)
                    {
                        referenceLevelEqualsZeroIDRef_ = referenceLevelEqualsZeroNode.Attributes["href"].Value;
                    }
                    else
                    {
                        referenceLevelEqualsZero_ = new XsdTypeBoolean(referenceLevelEqualsZeroNode);
                    }
                }
                else
                {
                    referenceLevelEqualsZero_ = new XsdTypeBoolean(referenceLevelEqualsZeroNode);
                }
            }
            
        
            XmlNode calculationDateNode = xmlNode.SelectSingleNode("calculationDate");
            
            if (calculationDateNode != null)
            {
                if (calculationDateNode.Attributes["href"] != null || calculationDateNode.Attributes["id"] != null) 
                {
                    if (calculationDateNode.Attributes["id"] != null) 
                    {
                        calculationDateIDRef_ = calculationDateNode.Attributes["id"].Value;
                        Period ob = new Period(calculationDateNode);
                        IDManager.SetID(calculationDateIDRef_, ob);
                    }
                    else if (calculationDateNode.Attributes["href"] != null)
                    {
                        calculationDateIDRef_ = calculationDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        calculationDate_ = new Period(calculationDateNode);
                    }
                }
                else
                {
                    calculationDate_ = new Period(calculationDateNode);
                }
            }
            
        
            XmlNode businessDaysNode = xmlNode.SelectSingleNode("businessDays");
            
            if (businessDaysNode != null)
            {
                if (businessDaysNode.Attributes["href"] != null || businessDaysNode.Attributes["id"] != null) 
                {
                    if (businessDaysNode.Attributes["id"] != null) 
                    {
                        businessDaysIDRef_ = businessDaysNode.Attributes["id"].Value;
                        BusinessCenter ob = new BusinessCenter(businessDaysNode);
                        IDManager.SetID(businessDaysIDRef_, ob);
                    }
                    else if (businessDaysNode.Attributes["href"] != null)
                    {
                        businessDaysIDRef_ = businessDaysNode.Attributes["href"].Value;
                    }
                    else
                    {
                        businessDays_ = new BusinessCenter(businessDaysNode);
                    }
                }
                else
                {
                    businessDays_ = new BusinessCenter(businessDaysNode);
                }
            }
            
        
            XmlNode dataCorrectionNode = xmlNode.SelectSingleNode("dataCorrection");
            
            if (dataCorrectionNode != null)
            {
                if (dataCorrectionNode.Attributes["href"] != null || dataCorrectionNode.Attributes["id"] != null) 
                {
                    if (dataCorrectionNode.Attributes["id"] != null) 
                    {
                        dataCorrectionIDRef_ = dataCorrectionNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(dataCorrectionNode);
                        IDManager.SetID(dataCorrectionIDRef_, ob);
                    }
                    else if (dataCorrectionNode.Attributes["href"] != null)
                    {
                        dataCorrectionIDRef_ = dataCorrectionNode.Attributes["href"].Value;
                    }
                    else
                    {
                        dataCorrection_ = new XsdTypeBoolean(dataCorrectionNode);
                    }
                }
                else
                {
                    dataCorrection_ = new XsdTypeBoolean(dataCorrectionNode);
                }
            }
            
        
            XmlNode correctionPeriodNode = xmlNode.SelectSingleNode("correctionPeriod");
            
            if (correctionPeriodNode != null)
            {
                if (correctionPeriodNode.Attributes["href"] != null || correctionPeriodNode.Attributes["id"] != null) 
                {
                    if (correctionPeriodNode.Attributes["id"] != null) 
                    {
                        correctionPeriodIDRef_ = correctionPeriodNode.Attributes["id"].Value;
                        Period ob = new Period(correctionPeriodNode);
                        IDManager.SetID(correctionPeriodIDRef_, ob);
                    }
                    else if (correctionPeriodNode.Attributes["href"] != null)
                    {
                        correctionPeriodIDRef_ = correctionPeriodNode.Attributes["href"].Value;
                    }
                    else
                    {
                        correctionPeriod_ = new Period(correctionPeriodNode);
                    }
                }
                else
                {
                    correctionPeriod_ = new Period(correctionPeriodNode);
                }
            }
            
        
            XmlNode maximumPaymentAmountNode = xmlNode.SelectSingleNode("maximumPaymentAmount");
            
            if (maximumPaymentAmountNode != null)
            {
                if (maximumPaymentAmountNode.Attributes["href"] != null || maximumPaymentAmountNode.Attributes["id"] != null) 
                {
                    if (maximumPaymentAmountNode.Attributes["id"] != null) 
                    {
                        maximumPaymentAmountIDRef_ = maximumPaymentAmountNode.Attributes["id"].Value;
                        NonNegativeMoney ob = new NonNegativeMoney(maximumPaymentAmountNode);
                        IDManager.SetID(maximumPaymentAmountIDRef_, ob);
                    }
                    else if (maximumPaymentAmountNode.Attributes["href"] != null)
                    {
                        maximumPaymentAmountIDRef_ = maximumPaymentAmountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        maximumPaymentAmount_ = new NonNegativeMoney(maximumPaymentAmountNode);
                    }
                }
                else
                {
                    maximumPaymentAmount_ = new NonNegativeMoney(maximumPaymentAmountNode);
                }
            }
            
        
            XmlNode maximumTransactionPaymentAmountNode = xmlNode.SelectSingleNode("maximumTransactionPaymentAmount");
            
            if (maximumTransactionPaymentAmountNode != null)
            {
                if (maximumTransactionPaymentAmountNode.Attributes["href"] != null || maximumTransactionPaymentAmountNode.Attributes["id"] != null) 
                {
                    if (maximumTransactionPaymentAmountNode.Attributes["id"] != null) 
                    {
                        maximumTransactionPaymentAmountIDRef_ = maximumTransactionPaymentAmountNode.Attributes["id"].Value;
                        NonNegativeMoney ob = new NonNegativeMoney(maximumTransactionPaymentAmountNode);
                        IDManager.SetID(maximumTransactionPaymentAmountIDRef_, ob);
                    }
                    else if (maximumTransactionPaymentAmountNode.Attributes["href"] != null)
                    {
                        maximumTransactionPaymentAmountIDRef_ = maximumTransactionPaymentAmountNode.Attributes["href"].Value;
                    }
                    else
                    {
                        maximumTransactionPaymentAmount_ = new NonNegativeMoney(maximumTransactionPaymentAmountNode);
                    }
                }
                else
                {
                    maximumTransactionPaymentAmount_ = new NonNegativeMoney(maximumTransactionPaymentAmountNode);
                }
            }
            
        
            XmlNode roundingNode = xmlNode.SelectSingleNode("rounding");
            
            if (roundingNode != null)
            {
                if (roundingNode.Attributes["href"] != null || roundingNode.Attributes["id"] != null) 
                {
                    if (roundingNode.Attributes["id"] != null) 
                    {
                        roundingIDRef_ = roundingNode.Attributes["id"].Value;
                        Rounding ob = new Rounding(roundingNode);
                        IDManager.SetID(roundingIDRef_, ob);
                    }
                    else if (roundingNode.Attributes["href"] != null)
                    {
                        roundingIDRef_ = roundingNode.Attributes["href"].Value;
                    }
                    else
                    {
                        rounding_ = new Rounding(roundingNode);
                    }
                }
                else
                {
                    rounding_ = new Rounding(roundingNode);
                }
            }
            
        
        }
        
    
        #region SettlementLevel_
        private WeatherSettlementLevelEnum settlementLevel_;
        public WeatherSettlementLevelEnum SettlementLevel_
        {
            get
            {
                if (this.settlementLevel_ != null)
                {
                    return this.settlementLevel_; 
                }
                else if (this.settlementLevelIDRef_ != null)
                {
                    settlementLevel_ = IDManager.getID(settlementLevelIDRef_) as WeatherSettlementLevelEnum;
                    return this.settlementLevel_; 
                }
                else
                {
                      return this.settlementLevel_; 
                }
            }
            set
            {
                if (this.settlementLevel_ != value)
                {
                    this.settlementLevel_ = value;
                }
            }
        }
        #endregion
        
        public string settlementLevelIDRef_ { get; set; }
        #region ReferenceLevelEqualsZero_
        private XsdTypeBoolean referenceLevelEqualsZero_;
        public XsdTypeBoolean ReferenceLevelEqualsZero_
        {
            get
            {
                if (this.referenceLevelEqualsZero_ != null)
                {
                    return this.referenceLevelEqualsZero_; 
                }
                else if (this.referenceLevelEqualsZeroIDRef_ != null)
                {
                    referenceLevelEqualsZero_ = IDManager.getID(referenceLevelEqualsZeroIDRef_) as XsdTypeBoolean;
                    return this.referenceLevelEqualsZero_; 
                }
                else
                {
                      return this.referenceLevelEqualsZero_; 
                }
            }
            set
            {
                if (this.referenceLevelEqualsZero_ != value)
                {
                    this.referenceLevelEqualsZero_ = value;
                }
            }
        }
        #endregion
        
        public string referenceLevelEqualsZeroIDRef_ { get; set; }
        #region CalculationDate_
        private Period calculationDate_;
        public Period CalculationDate_
        {
            get
            {
                if (this.calculationDate_ != null)
                {
                    return this.calculationDate_; 
                }
                else if (this.calculationDateIDRef_ != null)
                {
                    calculationDate_ = IDManager.getID(calculationDateIDRef_) as Period;
                    return this.calculationDate_; 
                }
                else
                {
                      return this.calculationDate_; 
                }
            }
            set
            {
                if (this.calculationDate_ != value)
                {
                    this.calculationDate_ = value;
                }
            }
        }
        #endregion
        
        public string calculationDateIDRef_ { get; set; }
        #region BusinessDays_
        private BusinessCenter businessDays_;
        public BusinessCenter BusinessDays_
        {
            get
            {
                if (this.businessDays_ != null)
                {
                    return this.businessDays_; 
                }
                else if (this.businessDaysIDRef_ != null)
                {
                    businessDays_ = IDManager.getID(businessDaysIDRef_) as BusinessCenter;
                    return this.businessDays_; 
                }
                else
                {
                      return this.businessDays_; 
                }
            }
            set
            {
                if (this.businessDays_ != value)
                {
                    this.businessDays_ = value;
                }
            }
        }
        #endregion
        
        public string businessDaysIDRef_ { get; set; }
        #region DataCorrection_
        private XsdTypeBoolean dataCorrection_;
        public XsdTypeBoolean DataCorrection_
        {
            get
            {
                if (this.dataCorrection_ != null)
                {
                    return this.dataCorrection_; 
                }
                else if (this.dataCorrectionIDRef_ != null)
                {
                    dataCorrection_ = IDManager.getID(dataCorrectionIDRef_) as XsdTypeBoolean;
                    return this.dataCorrection_; 
                }
                else
                {
                      return this.dataCorrection_; 
                }
            }
            set
            {
                if (this.dataCorrection_ != value)
                {
                    this.dataCorrection_ = value;
                }
            }
        }
        #endregion
        
        public string dataCorrectionIDRef_ { get; set; }
        #region CorrectionPeriod_
        private Period correctionPeriod_;
        public Period CorrectionPeriod_
        {
            get
            {
                if (this.correctionPeriod_ != null)
                {
                    return this.correctionPeriod_; 
                }
                else if (this.correctionPeriodIDRef_ != null)
                {
                    correctionPeriod_ = IDManager.getID(correctionPeriodIDRef_) as Period;
                    return this.correctionPeriod_; 
                }
                else
                {
                      return this.correctionPeriod_; 
                }
            }
            set
            {
                if (this.correctionPeriod_ != value)
                {
                    this.correctionPeriod_ = value;
                }
            }
        }
        #endregion
        
        public string correctionPeriodIDRef_ { get; set; }
        #region MaximumPaymentAmount_
        private NonNegativeMoney maximumPaymentAmount_;
        public NonNegativeMoney MaximumPaymentAmount_
        {
            get
            {
                if (this.maximumPaymentAmount_ != null)
                {
                    return this.maximumPaymentAmount_; 
                }
                else if (this.maximumPaymentAmountIDRef_ != null)
                {
                    maximumPaymentAmount_ = IDManager.getID(maximumPaymentAmountIDRef_) as NonNegativeMoney;
                    return this.maximumPaymentAmount_; 
                }
                else
                {
                      return this.maximumPaymentAmount_; 
                }
            }
            set
            {
                if (this.maximumPaymentAmount_ != value)
                {
                    this.maximumPaymentAmount_ = value;
                }
            }
        }
        #endregion
        
        public string maximumPaymentAmountIDRef_ { get; set; }
        #region MaximumTransactionPaymentAmount_
        private NonNegativeMoney maximumTransactionPaymentAmount_;
        public NonNegativeMoney MaximumTransactionPaymentAmount_
        {
            get
            {
                if (this.maximumTransactionPaymentAmount_ != null)
                {
                    return this.maximumTransactionPaymentAmount_; 
                }
                else if (this.maximumTransactionPaymentAmountIDRef_ != null)
                {
                    maximumTransactionPaymentAmount_ = IDManager.getID(maximumTransactionPaymentAmountIDRef_) as NonNegativeMoney;
                    return this.maximumTransactionPaymentAmount_; 
                }
                else
                {
                      return this.maximumTransactionPaymentAmount_; 
                }
            }
            set
            {
                if (this.maximumTransactionPaymentAmount_ != value)
                {
                    this.maximumTransactionPaymentAmount_ = value;
                }
            }
        }
        #endregion
        
        public string maximumTransactionPaymentAmountIDRef_ { get; set; }
        #region Rounding_
        private Rounding rounding_;
        public Rounding Rounding_
        {
            get
            {
                if (this.rounding_ != null)
                {
                    return this.rounding_; 
                }
                else if (this.roundingIDRef_ != null)
                {
                    rounding_ = IDManager.getID(roundingIDRef_) as Rounding;
                    return this.rounding_; 
                }
                else
                {
                      return this.rounding_; 
                }
            }
            set
            {
                if (this.rounding_ != value)
                {
                    this.rounding_ = value;
                }
            }
        }
        #endregion
        
        public string roundingIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/WeatherLegCalculation.cpp
// WeatherLegCalculation.cpp 
#include "WeatherLegCalculation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

WeatherLegCalculation::WeatherLegCalculation(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //settlementLevelNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* settlementLevelNode = xmlNode->FirstChildElement("settlementLevel");

   if(settlementLevelNode){settlementLevelIsNull_ = false;}
   else{settlementLevelIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- settlementLevelNode , address : " << settlementLevelNode << std::endl;
   #endif
   if(settlementLevelNode)
   {
      if(settlementLevelNode->Attribute("href") || settlementLevelNode->Attribute("id"))
      {
          if(settlementLevelNode->Attribute("id"))
          {
             settlementLevelIDRef_ = settlementLevelNode->Attribute("id");
             settlementLevel_ = boost::shared_ptr<WeatherSettlementLevelEnum>(new WeatherSettlementLevelEnum(settlementLevelNode));
             settlementLevel_->setID(settlementLevelIDRef_);
             IDManager::instance().setID(settlementLevelIDRef_,settlementLevel_);
          }
          else if(settlementLevelNode->Attribute("href")) { settlementLevelIDRef_ = settlementLevelNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { settlementLevel_ = boost::shared_ptr<WeatherSettlementLevelEnum>(new WeatherSettlementLevelEnum(settlementLevelNode));}
   }

   //referenceLevelEqualsZeroNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* referenceLevelEqualsZeroNode = xmlNode->FirstChildElement("referenceLevelEqualsZero");

   if(referenceLevelEqualsZeroNode){referenceLevelEqualsZeroIsNull_ = false;}
   else{referenceLevelEqualsZeroIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- referenceLevelEqualsZeroNode , address : " << referenceLevelEqualsZeroNode << std::endl;
   #endif
   if(referenceLevelEqualsZeroNode)
   {
      if(referenceLevelEqualsZeroNode->Attribute("href") || referenceLevelEqualsZeroNode->Attribute("id"))
      {
          if(referenceLevelEqualsZeroNode->Attribute("id"))
          {
             referenceLevelEqualsZeroIDRef_ = referenceLevelEqualsZeroNode->Attribute("id");
             referenceLevelEqualsZero_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(referenceLevelEqualsZeroNode));
             referenceLevelEqualsZero_->setID(referenceLevelEqualsZeroIDRef_);
             IDManager::instance().setID(referenceLevelEqualsZeroIDRef_,referenceLevelEqualsZero_);
          }
          else if(referenceLevelEqualsZeroNode->Attribute("href")) { referenceLevelEqualsZeroIDRef_ = referenceLevelEqualsZeroNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { referenceLevelEqualsZero_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(referenceLevelEqualsZeroNode));}
   }

   //calculationDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* calculationDateNode = xmlNode->FirstChildElement("calculationDate");

   if(calculationDateNode){calculationDateIsNull_ = false;}
   else{calculationDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- calculationDateNode , address : " << calculationDateNode << std::endl;
   #endif
   if(calculationDateNode)
   {
      if(calculationDateNode->Attribute("href") || calculationDateNode->Attribute("id"))
      {
          if(calculationDateNode->Attribute("id"))
          {
             calculationDateIDRef_ = calculationDateNode->Attribute("id");
             calculationDate_ = boost::shared_ptr<Period>(new Period(calculationDateNode));
             calculationDate_->setID(calculationDateIDRef_);
             IDManager::instance().setID(calculationDateIDRef_,calculationDate_);
          }
          else if(calculationDateNode->Attribute("href")) { calculationDateIDRef_ = calculationDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { calculationDate_ = boost::shared_ptr<Period>(new Period(calculationDateNode));}
   }

   //businessDaysNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* businessDaysNode = xmlNode->FirstChildElement("businessDays");

   if(businessDaysNode){businessDaysIsNull_ = false;}
   else{businessDaysIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- businessDaysNode , address : " << businessDaysNode << std::endl;
   #endif
   if(businessDaysNode)
   {
      if(businessDaysNode->Attribute("href") || businessDaysNode->Attribute("id"))
      {
          if(businessDaysNode->Attribute("id"))
          {
             businessDaysIDRef_ = businessDaysNode->Attribute("id");
             businessDays_ = boost::shared_ptr<BusinessCenter>(new BusinessCenter(businessDaysNode));
             businessDays_->setID(businessDaysIDRef_);
             IDManager::instance().setID(businessDaysIDRef_,businessDays_);
          }
          else if(businessDaysNode->Attribute("href")) { businessDaysIDRef_ = businessDaysNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { businessDays_ = boost::shared_ptr<BusinessCenter>(new BusinessCenter(businessDaysNode));}
   }

   //dataCorrectionNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* dataCorrectionNode = xmlNode->FirstChildElement("dataCorrection");

   if(dataCorrectionNode){dataCorrectionIsNull_ = false;}
   else{dataCorrectionIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- dataCorrectionNode , address : " << dataCorrectionNode << std::endl;
   #endif
   if(dataCorrectionNode)
   {
      if(dataCorrectionNode->Attribute("href") || dataCorrectionNode->Attribute("id"))
      {
          if(dataCorrectionNode->Attribute("id"))
          {
             dataCorrectionIDRef_ = dataCorrectionNode->Attribute("id");
             dataCorrection_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(dataCorrectionNode));
             dataCorrection_->setID(dataCorrectionIDRef_);
             IDManager::instance().setID(dataCorrectionIDRef_,dataCorrection_);
          }
          else if(dataCorrectionNode->Attribute("href")) { dataCorrectionIDRef_ = dataCorrectionNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { dataCorrection_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(dataCorrectionNode));}
   }

   //correctionPeriodNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* correctionPeriodNode = xmlNode->FirstChildElement("correctionPeriod");

   if(correctionPeriodNode){correctionPeriodIsNull_ = false;}
   else{correctionPeriodIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- correctionPeriodNode , address : " << correctionPeriodNode << std::endl;
   #endif
   if(correctionPeriodNode)
   {
      if(correctionPeriodNode->Attribute("href") || correctionPeriodNode->Attribute("id"))
      {
          if(correctionPeriodNode->Attribute("id"))
          {
             correctionPeriodIDRef_ = correctionPeriodNode->Attribute("id");
             correctionPeriod_ = boost::shared_ptr<Period>(new Period(correctionPeriodNode));
             correctionPeriod_->setID(correctionPeriodIDRef_);
             IDManager::instance().setID(correctionPeriodIDRef_,correctionPeriod_);
          }
          else if(correctionPeriodNode->Attribute("href")) { correctionPeriodIDRef_ = correctionPeriodNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { correctionPeriod_ = boost::shared_ptr<Period>(new Period(correctionPeriodNode));}
   }

   //maximumPaymentAmountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* maximumPaymentAmountNode = xmlNode->FirstChildElement("maximumPaymentAmount");

   if(maximumPaymentAmountNode){maximumPaymentAmountIsNull_ = false;}
   else{maximumPaymentAmountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- maximumPaymentAmountNode , address : " << maximumPaymentAmountNode << std::endl;
   #endif
   if(maximumPaymentAmountNode)
   {
      if(maximumPaymentAmountNode->Attribute("href") || maximumPaymentAmountNode->Attribute("id"))
      {
          if(maximumPaymentAmountNode->Attribute("id"))
          {
             maximumPaymentAmountIDRef_ = maximumPaymentAmountNode->Attribute("id");
             maximumPaymentAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(maximumPaymentAmountNode));
             maximumPaymentAmount_->setID(maximumPaymentAmountIDRef_);
             IDManager::instance().setID(maximumPaymentAmountIDRef_,maximumPaymentAmount_);
          }
          else if(maximumPaymentAmountNode->Attribute("href")) { maximumPaymentAmountIDRef_ = maximumPaymentAmountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { maximumPaymentAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(maximumPaymentAmountNode));}
   }

   //maximumTransactionPaymentAmountNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* maximumTransactionPaymentAmountNode = xmlNode->FirstChildElement("maximumTransactionPaymentAmount");

   if(maximumTransactionPaymentAmountNode){maximumTransactionPaymentAmountIsNull_ = false;}
   else{maximumTransactionPaymentAmountIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- maximumTransactionPaymentAmountNode , address : " << maximumTransactionPaymentAmountNode << std::endl;
   #endif
   if(maximumTransactionPaymentAmountNode)
   {
      if(maximumTransactionPaymentAmountNode->Attribute("href") || maximumTransactionPaymentAmountNode->Attribute("id"))
      {
          if(maximumTransactionPaymentAmountNode->Attribute("id"))
          {
             maximumTransactionPaymentAmountIDRef_ = maximumTransactionPaymentAmountNode->Attribute("id");
             maximumTransactionPaymentAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(maximumTransactionPaymentAmountNode));
             maximumTransactionPaymentAmount_->setID(maximumTransactionPaymentAmountIDRef_);
             IDManager::instance().setID(maximumTransactionPaymentAmountIDRef_,maximumTransactionPaymentAmount_);
          }
          else if(maximumTransactionPaymentAmountNode->Attribute("href")) { maximumTransactionPaymentAmountIDRef_ = maximumTransactionPaymentAmountNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { maximumTransactionPaymentAmount_ = boost::shared_ptr<NonNegativeMoney>(new NonNegativeMoney(maximumTransactionPaymentAmountNode));}
   }

   //roundingNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* roundingNode = xmlNode->FirstChildElement("rounding");

   if(roundingNode){roundingIsNull_ = false;}
   else{roundingIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- roundingNode , address : " << roundingNode << std::endl;
   #endif
   if(roundingNode)
   {
      if(roundingNode->Attribute("href") || roundingNode->Attribute("id"))
      {
          if(roundingNode->Attribute("id"))
          {
             roundingIDRef_ = roundingNode->Attribute("id");
             rounding_ = boost::shared_ptr<Rounding>(new Rounding(roundingNode));
             rounding_->setID(roundingIDRef_);
             IDManager::instance().setID(roundingIDRef_,rounding_);
          }
          else if(roundingNode->Attribute("href")) { roundingIDRef_ = roundingNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { rounding_ = boost::shared_ptr<Rounding>(new Rounding(roundingNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<WeatherSettlementLevelEnum> WeatherLegCalculation::getSettlementLevel()
{
   if(!this->settlementLevelIsNull_){
        if(settlementLevelIDRef_ != ""){
             return boost::shared_static_cast<WeatherSettlementLevelEnum>(IDManager::instance().getID(settlementLevelIDRef_));
        }else{
             return this->settlementLevel_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<WeatherSettlementLevelEnum>();
   }
}
boost::shared_ptr<XsdTypeBoolean> WeatherLegCalculation::getReferenceLevelEqualsZero()
{
   if(!this->referenceLevelEqualsZeroIsNull_){
        if(referenceLevelEqualsZeroIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(referenceLevelEqualsZeroIDRef_));
        }else{
             return this->referenceLevelEqualsZero_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<Period> WeatherLegCalculation::getCalculationDate()
{
   if(!this->calculationDateIsNull_){
        if(calculationDateIDRef_ != ""){
             return boost::shared_static_cast<Period>(IDManager::instance().getID(calculationDateIDRef_));
        }else{
             return this->calculationDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Period>();
   }
}
boost::shared_ptr<BusinessCenter> WeatherLegCalculation::getBusinessDays()
{
   if(!this->businessDaysIsNull_){
        if(businessDaysIDRef_ != ""){
             return boost::shared_static_cast<BusinessCenter>(IDManager::instance().getID(businessDaysIDRef_));
        }else{
             return this->businessDays_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<BusinessCenter>();
   }
}
boost::shared_ptr<XsdTypeBoolean> WeatherLegCalculation::getDataCorrection()
{
   if(!this->dataCorrectionIsNull_){
        if(dataCorrectionIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(dataCorrectionIDRef_));
        }else{
             return this->dataCorrection_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<Period> WeatherLegCalculation::getCorrectionPeriod()
{
   if(!this->correctionPeriodIsNull_){
        if(correctionPeriodIDRef_ != ""){
             return boost::shared_static_cast<Period>(IDManager::instance().getID(correctionPeriodIDRef_));
        }else{
             return this->correctionPeriod_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Period>();
   }
}
boost::shared_ptr<NonNegativeMoney> WeatherLegCalculation::getMaximumPaymentAmount()
{
   if(!this->maximumPaymentAmountIsNull_){
        if(maximumPaymentAmountIDRef_ != ""){
             return boost::shared_static_cast<NonNegativeMoney>(IDManager::instance().getID(maximumPaymentAmountIDRef_));
        }else{
             return this->maximumPaymentAmount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NonNegativeMoney>();
   }
}
boost::shared_ptr<NonNegativeMoney> WeatherLegCalculation::getMaximumTransactionPaymentAmount()
{
   if(!this->maximumTransactionPaymentAmountIsNull_){
        if(maximumTransactionPaymentAmountIDRef_ != ""){
             return boost::shared_static_cast<NonNegativeMoney>(IDManager::instance().getID(maximumTransactionPaymentAmountIDRef_));
        }else{
             return this->maximumTransactionPaymentAmount_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<NonNegativeMoney>();
   }
}
boost::shared_ptr<Rounding> WeatherLegCalculation::getRounding()
{
   if(!this->roundingIsNull_){
        if(roundingIDRef_ != ""){
             return boost::shared_static_cast<Rounding>(IDManager::instance().getID(roundingIDRef_));
        }else{
             return this->rounding_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Rounding>();
   }
}
}

